client/app/components/chat/ChatMessage.tsx

```diff
@@ -20,22 +26,22 @@ const Message: React.FC<MessageProps> = ({ messageContent, time, author }) => {
   // Text should have a different color to contrast with the background color
   const authorStyleProps = {
     ...styles.authorStyle,
-    color: settings && settings.theme !== "light" ? "white" : "black",
+    color: textSettings && textSettings.theme !== "light" ? "white" : "black",
   };
   return (
     <View style={styles.container}>
       <View style={styles.profileImageContainer}>
         <Image
-          style={styles.profileImage}
-          source={require("../../../assets/icons/user/fake_pfp.jpg")}
+          style={[styles.profileImage, { backgroundColor: profileSettings.profileColor }]}
+          source={icons[profileSettings.profilePicIndex]}
         />
       </View>
       <View style={styles.contentContainer}>
         <View style={styles.messageHeader}>
-          <Text style={authorStyleProps}>{author}</Text>
+          <Text style={authorStyleProps}>{profileSettings.displayName}</Text>
```

Comment:
If you change this prop to point to the display name, then every message will just show that it was written by you, rather than the person that actually sent the message. Just leave it as "author"

Reply:
Understood! I'll work on these changes now.